pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -275,22 +328,22 @@ func (srv PrometheusSrv) RouteGetRuleStatuses(c *contextmodel.ReqContext) respon
 
 	rulesTotals := make(map[string]int64, len(groupedRules))
 	for groupKey, rules := range groupedRules {
-		folder := namespaceMap[groupKey.NamespaceUID]
-		if folder == nil {
-			srv.log.Warn("Query returned rules that belong to folder the user does not have access to. All rules that belong to that namespace will not be added to the response", "folder_uid", groupKey.NamespaceUID)
+		folder, ok := opts.Namespaces[groupKey.NamespaceUID]
+		if !ok {
+			log.Warn("Query returned rules that belong to folder the user does not have access to. All rules that belong to that namespace will not be added to the response", "folder_uid", groupKey.NamespaceUID)
 			continue
 		}
-		ok, err := srv.authz.HasAccessToRuleGroup(c.Req.Context(), c.SignedInUser, rules)
+		ok, err := opts.AuthorizeRuleGroup(rules)
 		if err != nil {
 			ruleResponse.DiscoveryBase.Status = "error"
 			ruleResponse.DiscoveryBase.Error = fmt.Sprintf("cannot authorize access to rule group: %s", err.Error())
 			ruleResponse.DiscoveryBase.ErrorType = apiv1.ErrServer
-			return response.JSON(ruleResponse.HTTPStatusCode(), ruleResponse)
+			return ruleResponse
 		}
 		if !ok {
 			continue
 		}
```

Comment:
I do not think this is the right change. Previously, it used `HasAccessToRuleGroup`  that returns `(bool,error)` where the first argument is `true` if user has access to the group and `false` if it does not have, and `error` is not nil only if there is a generic error (db access or something else). The new change  uses `AuthorizeRuleGroup` that returns error in both cases above, and therefore it will exit on the first group user does not have access to. This is regression.